commitizen/commands/commit.py

```diff
@@ -73,14 +73,15 @@ def prompt_commit_questions(self) -> str:
         return message
 
     def __call__(self):
+        allow_empty: bool = self.arguments.get("allow_empty")
         dry_run: bool = self.arguments.get("dry_run")
         write_message_to_file: bool = self.arguments.get("write_message_to_file")
 
         is_all: bool = self.arguments.get("all")
         if is_all:
             c = git.add("-u")
 
-        if git.is_staging_clean() and not dry_run:
+        if git.is_staging_clean() and not (dry_run or allow_empty):
```

Comment:
If that's the case, we probably can check whether `--allow-empty` exists after `--` instead of adding a new argument. WDYT?

Reply:
I'll implement `-- --allow-empty detection` tomorrow :+1:.

Against the regular `--allow-empty` then, or can we keep the commonly used feature ?
If so, I'll rework the commits and restart new.

Reply:
I'm -0 on these additional arguments because they reduce maintainability. If we can extract all these arguments and the extra_cli_args logic and make `--allow-empty` an alias to `--—-allow-empty`, I think I'll be okay with it. 

Would like to know whether you think it's doable 🙂 Thanks!

Reply:
Ok fine with me, phasing out of `--allow-empty` feature, and will deprecate `-s` usage too :+1:

As long as `always_signoff` configuration remains working, and maybe one day `always_signgpg` as mentioned.

Reply:
Yep, we should definitly keep `always_signoff` config. If we want to have `always_signgpg`, maybe we can create an issue for tracking this 🙂

Reply:
For GPG, it went through my checklist and I commented here : 
* https://github.com/commitizen-tools/commitizen/issues/575
* https://github.com/commitizen-tools/commitizen/issues/347

Reply:
Done